docs/sql/statements/insert.rst

```diff
@@ -24,24 +24,31 @@ Synopsis
       { VALUES ( expression [, ...] ) [, ...] | ( query ) | query }
       [ ON CONFLICT (column_ident [, ...]) DO UPDATE SET { column_ident = expression [, ...] } |
         ON CONFLICT [ ( column_ident [, ...] ) ] DO NOTHING ]
+      [ RETURNING { * | output_expression [ [ AS ] output_name ] | relation.* } [, ...] ]
```

Comment:
What does `| relation.* ` stand for, can you give me an example please? Maybe its just me ;)

Reply:
I'm a little out of my depth here, but:

is the `foo.*` syntax valid EBNF? as a rule, I think our SQL syntax examples should be formatted as syntactically correct EBNF

if you generate EBNF directly from the Antlr definition, what would it look like for this bit?

Reply:
`relation.*` means that you can use the following syntax `insert into foo (id, name) values (1, 'bob') returning foo.*` returning id and name as result.

Reply:
@nomi sounds good. I think this is a bit inconsistent over the docs. Can you make a proposal how you would like to see it ?

Reply:
@mkleen it is inconsistent, yeah :( this is something I've been wanting to work on for a while. ideally, I want to generate the EBNF output into the docs (including railroad diagrams) directly from the Antlr definition

but until I get a chance to work on that, I don't know enough to have any preferences re how to format this. only that I think we should try to at least make sure of the following:

1) the SQL syntax definitions in the docs are kept up-to-date with our Antlr configuration (sounds obvious but I suspect there are a few cases where there's slight mismatch)

2) the SQL syntax definitions are valid EBNF

I don't know enough about EBNF at the moment to know whether `foo.*` is valid EBNF, and if it's not, what you would replace it with. sorry!

my guess, from this specific example, is that `foo.*` type expressions are already covered in this syntax definition by the `output_expression` parameter. so perhaps instead of adding `| relation.*` you could clarify in the text that `output_expression` supports single column names as well as globbed column names? not sure

Reply:
Originally we put `relation.*`  in update to the returning clause documentation https://github.com/crate/crate/blob/master/docs/sql/statements/update.rst
At the moment i would like to keep it consistent with the returning clause in update, and this pr is about insert. So maybe lets do another pr after cleaning up update and insert together ? Would that be ok for you ?

Reply:
yup sure ok